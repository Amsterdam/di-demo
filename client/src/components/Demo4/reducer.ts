const reducer = (state, action) => {
  console.log('reducer', state, action);

  switch (action.type) {
    case 'validateForm':
      return {
        ...state,
        formValid: true,
        owner: action.payload.owner
      };

    case 'invalidateForm':
      return {
        ...state,
        formValid: false,
      };

    case 'setProperties':
      return {
        ...state,
        hasResult: true,
        name: action.payload.name,
        street: action.payload.street,
        city: action.payload.city,
        telephone: action.payload.telephone,
      };

    default:
      return state;
  }
};

export default reducer;