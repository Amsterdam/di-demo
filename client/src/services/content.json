{
    "footer": {
        "column1": {
            "title": "Contact",
            "body": "Hebt u een vraag en kunt u het antwoord niet vinden op deze website? Neem dan contact met ons op."
        },
        "column2": {
            "title": "Volg de Gemeente",
            "body": [
                {
                    "label": "Lorem ipsum dolor sit.",
                    "url": "/"
                },
                {
                    "label": "Lorem.",
                    "url": "/"
                },
                {
                    "label": "Lorem ipsum.",
                    "url": "/"
                }
            ]
        },
        "column3": {
            "title": "Uit in Amsterdam",
            "body": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi dolor doloremque ea eos facere hic ipsum nobis provident quidem voluptates."
        },
        "bottom": [
            {
                "label": "Privacy and cookies",
                "url": "/"
            },
            {
                "label": "About this site",
                "url": "/"
            }
        ]
    }
}
