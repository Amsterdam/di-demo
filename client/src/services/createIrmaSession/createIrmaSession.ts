import axios from 'axios';
import * as irma from '@privacybydesign/irmajs';

// Types
interface IIrmaServerConfig {
    requestorname: string;
    uuid: string;
    irma: string;
    nodeUrl: string;
    docroot: string;
    port: number;
}

const port = 8000;

const instance = axios.create({
    baseURL: `${window.location.protocol}//${window.location.hostname}:${port}`
});

let config: IIrmaServerConfig;

export const getConfig = async (): Promise<IIrmaServerConfig> => {
    const response = await instance.get('/config');
    return response.data;
};

export const isMobile = (): boolean => {
    return /Android/i.test(window.navigator.userAgent) || /iPhone/.test(window.navigator.userAgent);
};

const createIrmaSession = async (dataType: string, holderElementId: string): Promise<unknown> => {
    if (!config) {
        config = await getConfig();
    }

    const irmaResponse = await instance.get(`/getsession/${dataType}`);
    const session = await irmaResponse.data;

    const { sessionPtr, token } = session;
    const sessionOptions = {
        method: 'canvas',
        element: holderElementId,
        showConnectedIcon: true,
        server: config.irma,
        token,
        language: 'nl',
        disableMobile: true
    };

    // TODO: Fix mobile session options
    // if (isMobile()) {
    //     sessionOptions.method = 'mobile';
    //     sessionOptions.disableMobile = false;
    // }

    const result = await irma.handleSession(sessionPtr, sessionOptions);
    const data = result.disclosed[0].reduce((acc, { id, rawvalue }) => ({ ...acc, [id]: rawvalue }), {});
    return data;
};

export default createIrmaSession;
