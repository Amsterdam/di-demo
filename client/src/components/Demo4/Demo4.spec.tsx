import React from 'react';
import { screen, act, fireEvent } from '@testing-library/react';
import { setupMocks, wrappedRender } from '@test/utils';
import Demo4 from '@components/Demo4/Demo4';
import createIrmaSession from '@services/createIrmaSession';
import content from '@services/content-nl';

// Setup all the generic mocks
setupMocks();

// MockcreateIrmaSession
jest.mock('@services/createIrmaSession');

describe('Demo4', () => {
    it('should render the initial header image', async () => {
        // Render demo 4
        await act(async (): Promise<any> => await wrappedRender(<Demo4 />));
        const headerImage: HTMLElement = await screen.findByTestId('headerImage');
        expect(headerImage).toMatchSnapshot();
    });

    it('should fail to start the IRMA flow when form is not filled', async () => {
        // Render demo 4
        await act(async (): Promise<any> => await wrappedRender(<Demo4 />));

        // Trigger IRMA flow
        const QRCodeButton = screen.getByTestId('qrCodeButton');
        await act(async (): Promise<any> => await fireEvent.click(QRCodeButton));

        // Expect modal not to be shown
        const QRCodeModal = screen.queryByTestId('qrCodeModal');
        expect(QRCodeModal).not.toBeInTheDocument();
    });

    it('should update the page after completing the IRMA flow', async () => {
        // Adjust mocked CreateIrmaSession to return a correct credentials
        const mockedCreateIrmaSession = createIrmaSession as jest.Mock<unknown>;
        mockedCreateIrmaSession.mockReturnValue({
            fullname: 'Test test',
            street: 'Teststraat',
            houseNumber: 100,
            zipcode: '1011PT',
            city: 'Amsterdam',
            mobilenumber: '0612345678',
            email: 'test@test.com'
        });

        // Render demo 4
        await act(async (): Promise<any> => await wrappedRender(<Demo4 />));

        // Fill in form
        const radioOption = screen.getByLabelText(content.demo4.form.optionYes);
        act(() => {
            fireEvent.click(radioOption);
        });

        // Trigger IRMA flow
        const QRCodeButton = screen.getByTestId('qrCodeButton');
        await act(async (): Promise<any> => await fireEvent.click(QRCodeButton));

        // Check if header image is updated
        const headerImage = screen.getByTestId('headerImage');
        expect(headerImage).toMatchSnapshot();

        // Check if demo notification is not visible
        const demoNotification = screen.queryByTestId('demoNotification');
        expect(demoNotification).toBeNull();

        // Check if correct alert is shown
        const hasResultAlert = screen.getByTestId('hasResultAlert');
        expect(hasResultAlert).toMatchSnapshot();
    });

    it('should update the page after failing the IRMA flow', async () => {
        // Adjust mocked CreateIrmaSession to return a correct credentials
        const mockedCreateIrmaSession = createIrmaSession as jest.Mock<unknown>;
        mockedCreateIrmaSession.mockReturnValue(null);

        // Render demo 4
        await act(async (): Promise<any> => await wrappedRender(<Demo4 />));

        // Fill in form
        const radioOption = screen.getByLabelText(content.demo4.form.optionYes);
        act(() => {
            fireEvent.click(radioOption);
        });

        // Trigger IRMA flow
        const QRCodeButton = screen.getByTestId('qrCodeButton');
        await act(async (): Promise<any> => await fireEvent.click(QRCodeButton));

        // Check if header image is updated
        const headerImage = screen.getByTestId('headerImage');
        expect(headerImage).toMatchSnapshot();

        // Check if demo notification is not visible
        const demoNotification = screen.queryByTestId('demoNotification');
        expect(demoNotification).toBeNull();

        // Check if correct alert is shown
        const hasErrorAlert = screen.getByTestId('hasErrorAlert');
        expect(hasErrorAlert).toMatchSnapshot();
    });
});
