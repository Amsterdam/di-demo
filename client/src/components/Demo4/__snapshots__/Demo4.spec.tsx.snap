// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo4 should render the initial header image 1`] = `
.c0 {
  width: 100%;
  margin-bottom: 8px;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 16px;
}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media screen and (min-width:540px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:375px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (max-width:767px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<img
  alt="Foto van geveltuin in straat"
  class="c0 c1"
  data-testid="headerImage"
  src="/assets/geveltuin-940.jpg"
  srcset="/assets/geveltuin-290.jpg 580w, /assets/geveltuin-940.jpg 1880w"
/>
`;

exports[`Demo4 should update the page after completing the IRMA flow 1`] = `
.c0 {
  width: 100%;
  margin-bottom: 8px;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 16px;
}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media screen and (min-width:540px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:375px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (max-width:1199px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (max-width:767px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<img
  alt="Foto van geveltuin in straat"
  class="c0 c1"
  data-testid="headerImage"
  src="/assets/geveltuin-overzicht-940.jpg"
  srcset="/assets/geveltuin-overzicht-290.jpg 580w, /assets/geveltuin-overzicht-940.jpg 1880w"
/>
`;

exports[`Demo4 should update the page after completing the IRMA flow 2`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c4 > svg {
  width: 20px;
  height: 20px;
}

.c4 svg circle,
.c4 svg rect,
.c4 svg polygon,
.c4 svg path {
  fill: #ffffff;
}

.c11 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c12 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c13 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c14 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c16 .c3 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-left: auto;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.3s ease;
  -webkit-transition: transform 0.3s ease;
  transition: transform 0.3s ease;
}

.c5 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  margin: 0;
  color: #000000;
  font-weight: 400;
  font-size: 16px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 20px;
  margin-bottom: 20px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c9 {
  margin-top: 0;
  color: inherit;
}

.c7 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c17 .c3 svg circle,
.c17:hover .c3 svg circle,
.c17 .c3 svg rect,
.c17:hover .c3 svg rect,
.c17 .c3 svg polygon,
.c17:hover .c3 svg polygon,
.c17 .c3 svg path,
.c17:hover .c3 svg path {
  fill: #ffffff;
}

.c15 .c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c2 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  position: relative;
  width: 100%;
  padding: 16px;
  background-color: #E6E6E6;
}

.c10 {
  line-height: 1.5em;
  margin-top: 8px;
  margin-bottom: 8px;
}

.c1 {
  margin-bottom: 16px;
  pointer-events: none;
  background-color: #0c8836;
}

.c1 .alert-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 .alert-content h2 {
  font-size: 18px;
}

.c1 .alert-content .icon {
  margin-right: 16px;
  min-width: px;
  background-repeat: no-repeat;
}

.c1 .alert-content * {
  color: #ffffff;
}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:1024px) {
  .c5 {
    font-size: 24px;
    line-height: 30px;
  }
}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media screen and (min-width:540px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:375px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (max-width:1199px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (max-width:767px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div
  aria-live="polite"
  class="c0 c1"
  data-testid="hasResultAlert"
  id="alert-sc-kHNMoK c1"
  role="alert"
  tabindex="-1"
>
  <div
    class="c2"
  >
    <div
      class="alert-content"
    >
      <span
        class="c3 c4 icon"
        color="#ffffff"
      >
        <svg
          focusable="false"
          viewBox="0 0 32 32"
        >
          <path
            d="M12.216 27.016L0 14.168l2.91-2.77 9.346 9.837L29.129 4 32 6.8z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div>
        <h2
          class="c5 c6 c7"
        >
          Uw persoons- en contactgegevens
        </h2>
        <p
          class="c8 c9 c10"
        >
          U heeft een formulier ingevuld  met uw naam, adres en contactgegevens.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Demo4 should update the page after failing the IRMA flow 1`] = `
.c0 {
  width: 100%;
  margin-bottom: 8px;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 16px;
}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media screen and (min-width:540px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:375px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (max-width:767px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<img
  alt="Foto van geveltuin in straat"
  class="c0 c1"
  data-testid="headerImage"
  src="/assets/geveltuin-940.jpg"
  srcset="/assets/geveltuin-290.jpg 580w, /assets/geveltuin-940.jpg 1880w"
/>
`;

exports[`Demo4 should update the page after failing the IRMA flow 2`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 22px;
  height: 22px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c4 > svg {
  width: 22px;
  height: 22px;
}

.c11 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c12 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c13 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c14 .c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c16 .c3 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-left: auto;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.3s ease;
  -webkit-transition: transform 0.3s ease;
  transition: transform 0.3s ease;
}

.c5 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  margin: 0;
  color: #000000;
  font-weight: 400;
  font-size: 16px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 20px;
  margin-bottom: 20px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c9 {
  margin-top: 0;
  color: inherit;
}

.c7 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c17 .c3 svg circle,
.c17:hover .c3 svg circle,
.c17 .c3 svg rect,
.c17:hover .c3 svg rect,
.c17 .c3 svg polygon,
.c17:hover .c3 svg polygon,
.c17 .c3 svg path,
.c17:hover .c3 svg path {
  fill: #ffffff;
}

.c15 .c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c2 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  position: relative;
  width: 100%;
  padding: 16px;
  background-color: #E6E6E6;
}

.c10 {
  line-height: 1.5em;
  margin-top: 8px;
  margin-bottom: 8px;
}

.c1 {
  margin-bottom: 16px;
  pointer-events: none;
  background-color: #FEC813;
}

.c1 .alert-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 .alert-content h2 {
  font-size: 18px;
}

.c1 .alert-content .icon {
  margin-right: 16px;
  min-width: 22px;
  background-repeat: no-repeat;
}

.c1 .alert-content * {
  color: #000000;
}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:1024px) {
  .c5 {
    font-size: 24px;
    line-height: 30px;
  }
}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:1024px) {

}

@media screen and (min-width:540px) {

}

@media screen and (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) and (max-width:1199px) {

}

@media (min-width:700px) and (max-width:1023px) {

}

@media (min-width:414px) and (max-width:699px) {

}

@media (max-width:413px) {

}

@media screen and (min-width:540px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:375px) {

}

@media screen and (max-width:1199px) {

}

@media screen and (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (max-width:1199px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (max-width:767px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

@media (min-width:1200px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div
  aria-live="polite"
  class="c0 c1"
  data-testid="hasErrorAlert"
  id="alert-sc-kHNMoK c1"
  role="alert"
  tabindex="-1"
>
  <div
    class="c2"
  >
    <div
      class="alert-content"
    >
      <span
        class="c3 c4 icon"
        color=""
        size="22"
      >
        <svg
          focusable="false"
          viewBox="0 0 36 36"
        >
          <path
            d="M33.63 30.9H2.37L18 3zm-24.44-4h17.62L18 11.19z"
          />
          <path
            d="M17 16.99h2v4.95h-2zm0 5.95h2v1.95h-2z"
          />
        </svg>
      </span>
      <div>
        <h2
          class="c5 c6 c7"
        >
          Niets doorgegeven
        </h2>
        <p
          class="c8 c9 c10"
        >
          U heeft uw gegevens niet doorgegeven aan de demosite van de gemeente Amsterdam.
        </p>
      </div>
    </div>
  </div>
</div>
`;
